Ugrid whatsnew (#4236)
SciTools/iris: docs/src/whatsnew/latest.rst

   The highlights for this minor release of Iris include:

   * We've added support for UGRID meshes which can now be loaded and attached

>      to a cube.

Reviewer: Perhaps a reference to how big a project this has been? There must be users wondering why there have been so many PR's but still minimal engagement with their requests. Perhaps that's just my take though, feel free to disagree!

✨ Features
===========

#. `@bjlittle`_, `@pp-mo`_ and `@trexfeathers`_ added support for unstructured
   meshes, as described by `UGRID`_. This involved adding a data model (:pull:`3968`,

>    :pull:`4014`, :pull:`4036`, :pull:`4053`) and API (:pull:`4063`, :pull:`4064`),
>    and supporting representation (:pull:`4033`, :pull:`4054`) and loading
>    (:pull:`4058`) of data on meshes.

Reviewer: ```suggestion
   :pull:`4014`, :pull:`4027`, :pull:`4036`, :pull:`4053`) and API (:pull:`4063`,
   :pull:`4064`), and supporting representation (:pull:`4033`, :pull:`4054`)
   and loading (:pull:`4058`) of data on meshes.
```

>    (:pull:`4058`, :pull:`4074`) of data on meshes.

Reviewer: I'm impressed you've managed to pick out the PR's from the maelstrom! Not sure what your criteria were, but looking through [all `mesh-data-model` PR's](https://github.com/SciTools/iris/pulls?page=1&q=is%3Apr+base%3Amesh-data-model) (with some human knowledge thrown in), I'm wondering whether you should also include:

* #4057 
* #4049 
* #4037 
* #4027 - definitely this one
* #4014
* #3968

What do you think?

Author: These do look like they're worth mentioning. Just wondering for #4027, while this has a large footprint in terms of lines changed, it seems like it is more of a logistical change as a mergeback than an actual piece of work like, say, #4015, which it contains. I guess the benefit of #4027 is that it contains references to multiple PRs, though linking major pieces of work like #4015 seems like it would refer closer to the actual work.

Reviewer: Yeah I can see the confusion here. The `ng-vat-mesh-api` branch was used as a development branch for multiple developers to collaborate on a single addition. So the end output of `ng-vat-mesh-api` is the intended focus.

_As opposed to our regular use of branches as feature branches, where each PR is more 'publicly consumable'. None of this is helped by the heavy over-use of the term 'merge-back' to refer to branch merges for many **different** purposes._

Author: I've made some changes to the listed PRs on the principle that chosen PRs should represent a significant change to functionality (minor PRs, PRs focused on testing and internal changes are ignored).

   Most of this new API can be found in :mod:`iris.experimental.ugrid`. The key
   objects introduced are :class:`iris.experimental.ugrid.Mesh`,
   :class:`iris.experimental.ugrid.MeshCoord` and
   :obj:`iris.experimental.ugrid.PARSE_UGRID_ON_LOAD`.
   A :class:`iris.experimental.ugrid.Mesh` contains a full description of a UGRID

>    type mesh. A :class:`iris.experimental.ugrid.MeshCoord` is a coordinate that
>    references and represents a :class:`iris.experimental.ugrid.Mesh`, it connects
>    a :class:`iris.experimental.ugrid.Mesh` to a cube. Cubes are also given the

Reviewer: ```suggestion
   type mesh. :class:`iris.experimental.ugrid.MeshCoord`\ s are coordinates that
   reference and represent a :class:`iris.experimental.ugrid.Mesh` for use
   on a :class:`~iris.cube.Cube`. Cubes are also given the
```

   property :method property:`iris.cube.Cube.mesh` which returns a
   :class:`iris.experimental.ugrid.Mesh` if one is attached to the cube via a
   :class:`iris.experimental.ugrid.MeshCoord`. Finally, the context manager
   :obj:`iris.experimental.ugrid.PARSE_UGRID_ON_LOAD` provides a way to load

>    UGRID files so that cubes can be returned with a

Reviewer: Any references to `Cube`s should be cross-references too. Probably only short cross-references given the expectation that Iris users are familiar with `Cube`s.
